### PR TITLE
Add missing urls on io structs

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @steveklabnik

### CODE REVIEW

File: src/libstd/io/mod.rs [hunk 1]
```diff
@@ -814,19 +815,23 @@ pub trait Read {
...
 ///   'true sink'.
 ///
 /// Writers are intended to be composable with one another. Many implementors
-/// throughout `std::io` take and provide types which implement the `Write`
+/// throughout [`std::io`] take and provide types which implement the `Write`
 /// trait.
 ///
+/// [`write()`]: #tymethod.write
+/// [`flush()`]: #tymethod.flush
+/// [`std::io`]: ../
```
Comment (ollie27) [inline src/libstd/io/mod.rs]: This should be `` [`std::io`]: index.html ``.

Comment (GuillaumeGomez) [inline src/libstd/io/mod.rs]: Thanks!

**Comment by frewsxcv** [comment]
🎊 🔗 🎊 

@bors r+ rollup

**Pull request merged**